src/pubsub.c

```diff
@@ -603,20 +612,24 @@ void punsubscribeCommand(client *c) {
 
 /* This function wraps pubsubPublishMessage and also propagates the message to cluster.
  * Used by the commands PUBLISH/SPUBLISH and their respective module APIs.*/
-int pubsubPublishMessageAndPropagateToCluster(robj *channel, robj *message, int sharded) {
-    int receivers = pubsubPublishMessage(channel, message, sharded);
-    if (server.cluster_enabled) clusterPropagatePublish(channel, message, sharded);
+int pubsubPublishMessagesAndPropagateToCluster(robj *channel, robj **messages, int count, int sharded) {
+    int receivers = pubsubPublishMessages(channel, messages, count, sharded);
+    if (server.cluster_enabled) clusterPropagatePublish(channel, messages, count, sharded);
     return receivers;
 }
 
+int pubsubPublishMessageAndPropagateToCluster(robj *channel, robj *message, int sharded) {
+    return pubsubPublishMessagesAndPropagateToCluster(channel, &message, 1, sharded);
+}
+
 /* PUBLISH <channel> <message> */
 void publishCommand(client *c) {
     if (server.sentinel_mode) {
         sentinelPublishCommand(c);
         return;
     }
 
-    int receivers = pubsubPublishMessageAndPropagateToCluster(c->argv[1], c->argv[2], 0);
+    int receivers = pubsubPublishMessagesAndPropagateToCluster(c->argv[1], &c->argv[2], c->argc - 2, 0);
```

Comment:
Is argc always 3 here or do we actually allow multi-message publish?

We have not decided the syntax for this yet (same command or a different MPUBLISH command) so maybe it's better to not assume it will be the same command.

Reply:
We do not allow multiple and propogatepublish also does not allow multiple as of now

Reply:
So, more exlicit like this?

```suggestion
    int receivers = pubsubPublishMessagesAndPropagateToCluster(c->argv[1], &c->argv[2], 1, 0);
```

Reply:
Oh thats what you meant, okay

Reply:
Yeah, with `c->argc - 2` it looks like it would actually work with a variable number of messages.